### PR TITLE
feat: implement ceramic session

### PR DESCRIPTION
**Pull request by andyv09**
Masca:
- Added `ceramicSessionKey` to account state
- Added 2 new RPC methods; `setCeramicSessionKey` and `verifyStoredCeramicSessionKey`
- Changed Ceramic Client to start using session from state

Connector:
- Added new RPC methods to API
- Added function that checks if there's a valid sessionKey in Masca. If there isn't one it tries to create one using `personal_sign`.
- Added checks to each function that uses ceramic

Types:
- Added new types

### CODE REVIEW

File: packages/connector/src/snap.ts [hunk 1]
```diff
@@ -346,6 +398,30 @@ export async function sendOIDCAuthorizationResponse(
...
+      method: 'setCeramicSessionKey',
+      params: { sessionKey },
+    },
+    this.snapId
+  );
+}
+
+export async function verifyStoredCeramicSessionKey(
+  this: Masca
+): Promise<Result<string>> {
```
Comment (martines3000) [inline packages/connector/src/snap.ts:416]: Better to return `boolean`

File: packages/connector/package.json [hunk 2]
```diff
@@ -43,8 +43,10 @@
   "dependencies": {
     "@blockchain-lab-um/masca-types": "*",
     "@blockchain-lab-um/utils": "*",
+    "@didtools/pkh-ethereum": "0.0.2",
```
Comment (martines3000) [inline packages/connector/package.json:46]: Do we need this package ?

File: packages/connector/src/snap.ts [hunk 3]
```diff
@@ -266,6 +308,16 @@ export async function createVC(
   this: Masca,
   params: CreateVCRequestParams
 ): Promise<Result<VerifiableCredential>> {
+  if (
+    params.options?.save === true &&
+    (!params.options?.store || params.options?.store.includes('ceramic'))
+  ) {
+    const sessionSet = await verifyAndSetCeramicSession(this);
```
Comment (martines3000) [inline packages/connector/src/snap.ts:315]: In case `verifyAndSetCeramicSession` fails, we should return `ResultObject.error`

**Review by martines3000 — CHANGES_REQUESTED** [review]

File: packages/connector/package.json (hunk 2 above)
Comment (andyv09) [reply packages/connector/package.json:46]: Yes. Its used to for authorizing did session

File: packages/connector/src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+export async function verifyAndSetCeramicSession(masca: Masca) {
+  // Check if there is valid session in Masca
+  const api = masca.getMascaApi();
+  const session = await api.verifyStoredCeramicSessionKey();
+  if (!isError(session)) {
+    return true;
+  }
+
+  // Start new session if there is no valid session
+  const ethProvider = await detectEthereumProvider();
```
[the remark this answers is not in the document]
Comment (andyv09) [reply packages/connector/src/utils.ts:17]: Probably yes. I dont think you can get to this point without `window.ethereum` as it's needed to install Masca, etc.

File: packages/connector/src/snap.ts (hunk 3 above)
Comment (andyv09) [reply packages/connector/src/snap.ts:315]: We return ResultObject.error in `verifyAndSetCeramicSession()` which is then returned in `snap.ts` aswell

**Pull request merged**